_pytest/nodes.py

```diff
@@ -178,15 +179,23 @@ def add_marker(self, marker):
         elif not isinstance(marker, MarkDecorator):
             raise ValueError("is not a string or pytest.mark.* Marker")
         self.keywords[marker.name] = marker
+        self._markers.update([marker])
+
+    def find_markers(self, name):
+        """find all marks with the given name on the node and its parents
+
+        :param str name: name of the marker
+        """
+        for node in reversed(self.listchain()):
+            for mark in node._markers.find(name):
```

Comment:
Seems like I can only iterate when I know what I'm looking for.  Can we allow iterating over all marks as well?

Reply:
with the current mark mechanism you dont really get that sane iteration as well, so i dodnt want to add an api since apis are easy to get wrong

as for mamrkers - i would like to subscribe them to the "you better know what you are looking for" creed

i am currently against adding the arbitrary iteration unless someone can demonstrate a real use-case